xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -1037,6 +1047,7 @@ bool CXBMCApp::StartActivity(const std::string& package,
 
       if (e["type"] == "string")
         newIntent.putExtra(e["key"].GetString(), e["value"].GetString());
+        CLog::LogF(LOGDEBUG, "Putting extra key: {}, value: {}", e["key"].GetString(), e["value"].GetString());
```

Comment:
If you add a logline here, you need to use `{}` for the if.

Reply:
is there a linter enabled somewhere? Been switching between my chromebook and laptop with the development for testing with android. That kicks of the builds. Fixing it now, but indeed want to see it beforehand.

On another note, I still notice some issues with the intent parameter when invoking android apps. This is separate from the actual changes in this PR, but something I noticed already with your previous PR. If the intent is used it will throw exceptions. Perhaps I'll start up a different branch to see if we can incorporate "setComponent" to solve that part.
Starting up basic android apps still works as before though.

Reply:
You need to install git-clang-format as described in my other post.
Not sure which exception you refer to.

Reply:
The exception is not giving further details, except that it is triggered once the intent value is applied. Without it starts up the application as expected.
```
2022-09-21 15:30:15.699 T:2924    DEBUG <general>: StartActivity: package: com.retroarch
2022-09-21 15:30:15.699 T:2924    DEBUG <general>: StartActivity: intent: android.intent.action.MAIN
2022-09-21 15:30:15.701 T:2924    DEBUG <general>: StartActivity: dataType: 
2022-09-21 15:30:15.701 T:2924    DEBUG <general>: StartActivity: dataURI: 
2022-09-21 15:30:15.701 T:2924    DEBUG <general>: StartActivity: flags: 
2022-09-21 15:30:15.701 T:2924    DEBUG <general>: StartActivity: extras: [{"key": "ROM", "value": "'/storage/emulated/0/Download/work/*********.nes'", "type": "string"}, {"key": "LIBRETRO", "value": "/data/user/0/com.retroarch/cores/nestopia_libretro_android.so", "type": "string"}, {"key": "CONFIGFILE", "value": "/storage/emulated/0/Android/data/com.retroarch/files/retroarch.cfg", "type": "string"}]
2022-09-21 15:30:15.701 T:2924    DEBUG <general>: StartActivity: action: com.retroarch/.browser.retroactivity.RetroActivityFuture
2022-09-21 15:30:15.701 T:2924    DEBUG <general>: StartActivity: category: android.intent.category.LAUNCHER
2022-09-21 15:30:15.712 T:2924    ERROR <general>: StartActivity: ExceptionOccurred launching com.retroarch
```

Reply:
I think the exception should show up in logcat.
But you can add `xbmc_jnienv()->ExceptionDescribe();` between `xbmc_jnienv()->ExceptionCheck()` and `xbmc_jnienv()->ExceptionClear();` and it should show up in the log if I'm not mistaken.

Reply:
Installed clang-format by the way. Did the formatting, but it had quite some formatting changes, even in code I didnt touch. So curious if it is applied correctly now

Reply:
> Installed clang-format by the way. Did the formatting, but it had quite some formatting changes, even in code I didnt touch. 

That's why I referred to `git-clang-format` and not `clang-format` which only formats lines that you have changed.

Reply:
Alright, clear. I merged in the master branch again to update everything, getting an older merge commit in. It aborts the build now. Didn't see anything about it in the contribution documentation as far as I can see.
Should I revert the last commits?

Reply:
Ok so ususally we don't merge master back in our featurebranches but instead rebase so you never get merge commits in your feature branch.
I checked out your current state and here's what I suggest:
```
git pull --rebase upstream master
git rebase -i hash-of-first-commit^ # Note the ^ which is needed to include your first commit for the interactive rebase
```
You'll get an editor, where you will delete the line with the clang-changes commit.
For the very top commit you leave the keyword "pick" for all the others you change it to "f".
Then you save and quit the editor.
Your branch will then have only one commit.
Follow up with:
```
git reset HEAD^
git add xbmc/
git-clang-format
git add xbmc/
git commit
```

This will fix your branch so that you have one commit with all your changes perfectly formatted.

Reply:
Forgot one thing:
Once you have only one commit on your branch "don't pull again", instead you need to force-push your branch so that this PR will get updated with the correct version.

Reply:
Ah, overlooked your second post. Did the first and pushed that indeed.

Reply:
I should propably use git pull rebase more often. 
But yes didnt do the force push. Would it work if I made a minor change and commit,pushed that to trigger the PR update?

Reply:
Apparently something went wrong with your PR.
If you do a `git pull --rebase upstream master` your commits will be recommited on top of current master.
You must force push after that, a normal push will be rejected by git because your local branch doesn't match with your remote branch anymore.

Probably the easiest way to fix your work is to:
```
git fetch upstream
git reset --hard upstream/master
git cherry-pick bdf731d6f7c3feb3892a86e4ca69ba0e358d5708
```
You might want to change the author to yourself after picking my commit:
```
git commit --amend --author "chrisism <cwjungerius@gmail.com>"
```

Then force push your branch:
```
git push -f origin add-android-intent-parameters
```

Reply:
Mm, it cannot find that commit for the cherry pick (bad object). Did I need to redo the git pull -rebase before the git fetch?

Reply:
Ah sorry, that commit lives in my fork.
So you need to:
```
git remote add a1rwulf git@github.com/a1rwulf/xbmc.git
git fetch a1rwulf
```
Then you'll be able to pick it.

Reply:
Did a clean git pull rebase actually and fixed all conflicts with the force push. Seems to have worked.
I still have the component/class parameter to do, but I will start a clean feature branch for that and do it the correct way.
Lets finish this one. All reviewers need to approve?